glommio/src/io/directory.rs

```diff
@@ -68,6 +68,21 @@ impl Directory {
         Ok(Directory { file })
     }
 
+    /// Opens a file under this directory, returns a DMA file
+    ///
+    /// NOTE: Path must not contain directories and just be a file name
+    pub async fn open_file<P: AsRef<Path>>(&self, path: P) -> io::Result<DmaFile> {
+        if path.as_ref().components().count() > 1 {
```

Comment:
This error handling code should be split out into a non-async separate helper function, as this code is now present in at least 2 functions. Additionally, the code should check if the path contains *only* a single directory separator.

Reply:
👍 i added a function that returns a `bool` but wasn't sure if the convention was to return `Result<()>` instead so the caller can use `?` operator on it, lmk if i should change it